Dev front

bitBadger8 opened the pull request:
Latest Frontend Version

Ronan-UP committed:
Created new branch, added method to get the remaining allowance of a user.

Ronan-UP committed:
Added a method for payment to the smart contract, removed the method to get allowance in UnisonService, since it already exists in TokenService.

Ronan-UP committed:
Added a method to get a new message stream in MessageService.

Ronan-UP committed:
Added comment and attempted to make a custom Metamask request. This failed, and the test has been commented out in wallet.dart.

Ronan-UP committed:
Added initial version of HTTP PUT handler in backendAPI.dart.

Ronan-UP committed:
Added method to jury to get the raw vote, in stead of a string.

Ronan-UP committed:
voteNumber now updates correctly, asif I'm a competent coder now.

bitBadger8 committed:
Jury Voting is now responsive

bitBadger8 committed:
Updated TODO

Ronan-UP committed:
Added payingUser field to Contract.

Ronan-UP committed:
Added agreementState method in blockchainAgreement.dart.

bitBadger8 committed:
More Descriptive Blockchain status added to Agreement

Ronan-UP committed:
Changed agreementState to an enum.

bitBadger8 committed:
Fixed Red Screen on initial application launch

bitBadger8 committed:
Authentication Screen is reworked and restyled

bitBadger8 committed:
Added a custom loading animation to auth screen

bitBadger8 committed:
Updated Dashboard screen -new styles and Revised FAB

Ronan-UP committed:
Added explanation of agreement states.

Ronan-UP committed:
Updated tryAddUser in login service.

Ronan-UP committed:
Updated getInvolvedAgreements.

Ronan-UP committed:
Removed unnecessary method.

Ronan-UP committed:
Removed two more unnecessary methods.

Ronan-UP committed:
Updated saveAgreement and the Contract toJSON.

Ronan-UP committed:
Updated saveCondition and Condition toJSON.

Ronan-UP committed:
Updated accept/reject condition.

bitBadger8 committed:
General neaten up including responsiveness to add conditions buttons

bitBadger8 committed:
Merge remote-tracking branch 'origin/dev_front_improvements' into dev_front_improvements

Ronan-UP committed:
Re-added getAgreement to commonService.dart.

Ronan-UP committed:
Updated negotiationService.dart to new api standards.

Ronan-UP committed:
Updated message toJSON, sendMessage and _getMessageHandler,

Ronan-UP committed:
Updated messaging functions with new endpoints.

Ronan-UP committed:
Added display for current allowance on the dashboard, and the necessary changes to the services.

Ronan-UP committed:
Made the allowance button a bit more aesthetically pleasing.

bitBadger8 committed:
Updated Depreciated Package for typing Animation

bitBadger8 committed:
Updated Condition items
- added descriptions
- modified icons and coloring

bitBadger8 committed:
SoC implemented for Contract Conditions
- removed unnecessary implementations
- allowed expansion of form elements (for later usage)
- built separate functions for different types of conditions

bitBadger8 committed:
Added a flickering animation for no conditions and renamed negotiation service to be inline with coding standards

bitBadger8 committed:
SoC for Jdenticon of ContractItem

bitBadger8 committed:
SoC for FunkyText, general formatting

bitBadger8 committed:
Updated AnimationText Kit from 1.3.1-> 4.2.1

bitBadger8 committed:
Removed redundant screen (not used)

bitBadger8 committed:
SoC for Shaded Text (for fade effect)

bitBadger8 committed:
SoC for Jdenticon generation
- added color hash as an argument (for later use)

bitBadger8 committed:
Updated Primary Swatch from deepOrange to pink

bitBadger8 committed:
Updated animated text kit: 1.1.0->4.2.1

bitBadger8 committed:
Created fileversion backup - to be deleted later

bitBadger8 committed:
Complete rework of view contract screen
- focus on SoC
- removed redundant code and variables

bitBadger8 committed:
General cleanup of code for edit_contract_screen.dart

bitBadger8 committed:
Updated Create Agreement Screen and primary Swatch

Ronan-UP committed:
Added stream in messaging service fr testing purposes.

bitBadger8 committed:
Messages auto update - rework need (looks wise)

Ronan-UP committed:
Updated Message fromJSON and getMessageHandler.

Ronan-UP committed:
Updated judgeService to use new endpoints

Ronan-UP committed:
Added file for new ApiResponse class.

Ronan-UP committed:
Added implementation for ApiResponse.

Ronan-UP committed:
Added error handling to ApiResponse.

Ronan-UP committed:
BackendAPI methods now return ApiResponses with error messages in stead of throwing errors.

Ronan-UP committed:
Put requests also return an ApiResponse now, and improved error handling in ApiResponse.

Ronan-UP committed:
MessageService now uses ApiResponses.

bitBadger8 committed:
Pass checkers through to message panel

bitBadger8 committed:
Pass checkers through to message panel

bitBadger8 committed:
Update to return a message_item.dart

bitBadger8 committed:
Updated jdenticon_svg.dart to receive the hues as an argument

Ronan-UP committed:
NegotiationService now uses ApiResponses, removed unnecessary try/catch blocks.

Ronan-UP committed:
CommonService now uses ApiResponses, and removed unnecessary try/catch blocks. This implies that exception handling must be done by the UI.

Ronan-UP committed:
Removed unnecessary exception handling in MessageService.

bitBadger8 committed:
Message builder for sending and receiving items

bitBadger8 committed:
Added MessagePanel details panel

bitBadger8 committed:
Added Refresh Dashboard button

bitBadger8 committed:
Neat text widget for adding to blockchain

bitBadger8 committed:
Added participants panel to messaging screen

bitBadger8 committed:
Contract Action Area - working state

bitBadger8 committed:
Removed CreationSubscription causing infinite wait on blockchain response

bitBadger8 committed:
Commented out prints (needed cleaner console)

bitBadger8 committed:
Build an 'Accept Blockchain agreement when shouldAAccept is true'
- functionality added as well

bitBadger8 committed:
No more reload when sending message

bitBadger8 committed:
Extracted message input panel with logic

bitBadger8 committed:
Messaging input panel is now beautiful

Ronan-UP committed:
Added Contact.dart and contactService.dart.

Ronan-UP committed:
Renamed Contract.dart to contract.dart (lowercase).

Ronan-UP committed:
Updated misspelling of Contact, and added contactList.dart.

Ronan-UP committed:
Implemented fromJSON constructors for Contact and ContactList.

Ronan-UP committed:
Added optional body parameter to put requests. Continued implementation of ContactService.

Ronan-UP committed:
Re-implemented the BackendApi class, using function pointers.

Ronan-UP committed:
Changed the retry delay to 2 seconds in BackendAPI.

Ronan-UP committed:
Removed event-listening functionality, as it is no longer needed.

Ronan-UP committed:
Several minor fixes to the api services.

Ronan-UP committed:
Added more methods to ContactService and fixed Contact fromJSON.

Ronan-UP committed:
Minor updates to contactList. Added setContacts method to Contact.

Ronan-UP committed:
Added more comments.

Ronan-UP committed:
Added basic contact list screen and updated ContractList.fromJSON.

Ronan-UP committed:
The screen to display contact lists is fully operational. Plotting course to Alderaan now.

Ronan-UP committed:
Updated contact screen, made the tile purple.

bitBadger8 committed:
All Agreement States are accounted for and render the correct actions

bitBadger8 committed:
Removed Action logic, neatened up details and included action area widget

bitBadger8 committed:
Updated Todo list (sprint next week)

Ronan-UP committed:
Added ability to create a new contact list from the contact list screen.

Ronan-UP committed:
Fixed issue with contact service. Adding a user now properly passes in the body to the put request.

Ronan-UP committed:
The alertDialog for adding a contact now displays correctly.

Ronan-UP committed:
The contacts screen now allows a user to remove a contact from a contact list.

bitBadger8 committed:
Merge branch 'dev_front_improvements' into dev_front_improvements_api

Ronan-UP committed:
Merge pull request #86 from COS301-SE-2021/dev_front_improvements_api

Dev front improvements api

Ronan-UP committed:
Fixed issues due to strange merge.

Ronan-UP committed:
Updated readme.

Ronan-UP committed:
Merge pull request #87 from COS301-SE-2021/dev_front_improvements_api

Dev front improvements api

bitBadger8 committed:
Merge pull request #88 from COS301-SE-2021/dev_front_improvements

Dev front improvements

Ronan-UP committed:
Updated getInvolvedAgreements in judge service, it now uses ApiResponse.

bitBadger8 committed:
Merge branch 'dev' into dev_front

Kaleb-Bruwer commented on frontend_scv/lib/screens/view_contract_screen_backup.dart:
```diff
@@ -0,0 +1,328 @@
+// import 'package:flutter/material.dart';
```
Please delete this file

Kaleb-Bruwer commented on frontend_scv/lib/services/Server/commonService.dart:
```diff
@@ -1,125 +1,46 @@
 
-    List<dynamic> jsonList = ((response['agreements']));
-    //print("Fetched" + jsonList.toString());
+    print ('I am here....');
```
Fine for now, but please remove this before demo 4

bitBadger8 replied:
Will let Ronan as well

bitBadger8 committed:
Removed unnecessary file

This file was removed, it is no longer needed as reference for the refined version of this screen.
